backend/routes/event.js

```diff
@@ -1,125 +1,13 @@
-var express = require("express");
-var router = express.Router();
+const express = require("express");
+const router = express.Router();
 
 const auth = require("../util/auth");
-const Event = require("../models/event.model");
-
-// Creates a event
-router.post("/", auth, function (req, res) {
-  const event = new Event({
-    title: req.body.title,
-    community: req.body.community,
-    description: req.body.description,
-    pdfPath: req.body.pdfPath || null,
-    eventLink: req.body.eventLink || null,
-    eventDate: req.body.eventDate,
-    banner: req.body.banner || null,
-  });
-
-  event
-    .save()
-    .then((result) => {
-      res.status(201).json({
-        result: result,
-        message: "OK",
-      });
-      console.log(result);
-    })
-    .catch((err) => {
-      res.status(500).json({
-        message: "Something went wrong",
-      });
-      console.log(err);
-    });
-});
-
-// Returns 5 the Event (sorted : latest first)
-router.get("/", (req, res) => {
-  Event.find({})
-    .sort({ createdAt: -1 })
-    .limit(5)
-    .then((result) => {
-      res.status(200).json({
-        result: result,
-        message: "OK",
-      });
-    })
-    .catch((err) => {
-      console.log(err);
-      res.status(500).json({
-        message: "Something went wrong",
-      });
-    });
-});
-
-// Returns all the events
-router.get("/all", (req, res) => {
-  Event.find({})
-    .sort({ createdAt: -1 })
-    .then((result) => {
-      res.status(200).json({
-        result: result,
-        message: "OK",
-      });
-    })
-    .catch((err) => {
-      console.log(err);
-      res.status(500).json({
-        message: "Something went wrong",
-      });
-    });
-});
-
-// Deletes a Event
-router.delete("/:id", auth, (req, res, next) => {
-  Event.findOneAndRemove({ _id: req.params.id })
-    .then(() => {
-      res.status(200).json({
-        message: "OK",
-      });
-    })
-    .catch((err) => {
-      console.log(err);
-      res.status(500).json({
-        message: "Something went wrong",
-      });
-    });
-});
+const eventController = require("../controllers/eventController")
 
+router.post("/", auth, eventController.create_event);
+router.get("/", eventController.get_event);
```

Comment:
Add separate getters for `/all` where we retrive all the events and `/` where we retrieve only the first 5 when sorted in decreasing order according to `eventDate`.

Reply:
I have go through the preview code, where the event route for `/` and `/all` are trying to achieve the same result, differences for both them are just adding the `.limit(5`), where it violated the design principle DRY, 
So for current `router.get("/", eventController.get_event); `route,
we can pass the req.body.options, json object
`{ options: { limit: 5} }`
to achieve the same result 

*FYI in future for the frontend event table, it can make a paginate feature using this route options too

Reply:
That's correct, I got to learn about DRY today. Thanks!